### PR TITLE
LPS-177938 Implement FDS View persistence and UI for creation

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-177938 Implement FDS View persistence and UI for creation](https://issues.liferay.com/browse/LPS-177938)
- [Mockups](https://www.figma.com/file/VTOLTBFe99y9kA7AY3TcNR/lps-175276-design-app-to-manage-datasets?node-id=937%3A117695&t=KX193EH2qagQjOgV-0)

### What is the goal of this PR?

As the task name says, we are implementing FDS View persistence and UI for creation.

Notes:
- @gabrielwas please take a look at the backend part of PR. Is this the right way of using Objects API, specifically object relations with `ObjectRelationshipLocalService`? What is the purpose of the `parameterObjectFieldId`? I set it to 0 in the PR, as I've seen it set to 0 in some tests like [this one](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L312-L320).
- @kresimir-coko I did some squashing of duplicated code, to common constants and components. We can do more, for example to abstract saving of entity into a utility. I chose to leave it for a followup task, because that would be a utility, and we would want utilities to be covered by JEST tests. This PR already has large enough scope. 
- @ugeortiz In difference to mockups, I didn't add the `Provider` disabled field in the creation modal. It is not necessary for the base workflow. We discussed about this, but there was no strong conclusion. If it's valuable for UX, we can add this in a followup task.

### What does it look like?

https://user-images.githubusercontent.com/5435511/224302462-eaae1c5d-4764-463a-a7e7-2d94ab883e93.mp4

### Steps to reproduce
 

1. Add `feature.flag.LPS-164563=true` to `portal-ext.properties`
2. :warning: If you previously opened Datasets page, **drop database and create a new one**. This PR changes persistance layer.
3. Start server
4. Open Global Menu > Control Panel > Object > Datasets

### CODE REVIEW

**Comment by ugeortiz** [comment]
@markocikos the provider input was intended to give some kind of reference about the source of data you are using in the view you are creating. As we provide it in the Default screen, and long API names could create some confusion in this step, I think it is OK to remove it.

**Comment by markocikos** [comment]
We discussed the backend of this PR [in Slack](https://liferay.slack.com/archives/C046HH4TSUF/p1678450609343769). The conclusion that it is technically ok, we can forward it, but we should rewrite and improve it in a future task.

@kresimir-coko @dsanz Please take a look at the frontend part.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSViews.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		<>
+			<FrontendDataSet
+				apiURL={`${fdsEntriesAPIURL}/${fdsEntryId}/fdsViews`}
+				creationMenu={creationMenu}
+				id={`${namespace}FDSViews`}
+				style="fluid"
+				views={views}
+				{...PAGINATION_PROPS}
+			/>
+		</>
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSViews.tsx:232]: I think this empty fragment is obsolete.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/RequiredMark.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const RequiredMark = () => (
+	<>
+		<span className="inline-item-after reference-mark text-warning">
+			<ClayIcon symbol="asterisk" />
+		</span>
+		<span className="hide-accessible sr-only">
+			{Liferay.Language.get('required')}
+		</span>
+	</>
+);
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/RequiredMark.tsx:27]: We have that bug in our formatter so we need to manually add empty lines between sibling elements inside react fragments

**Review by kresimir-coko — CHANGES_REQUESTED** [review]
Couple nits but looks good

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSViews.tsx (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSViews.tsx]: I'll also make this change in `FDSEntries.tsx`. 

I could have sworn that this caused an SF error few days ago. Was this a bug in source formatter that we recently fixed? Maybe I was just imagining things :sweat_smile:

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
@kresimir-coko Does the :+1: in https://github.com/liferay-frontend/liferay-portal/pull/3113#issuecomment-1468111927 mean that the review is `Approved`?

**Comment by kresimir-coko** [comment]
Yes, sorry!

**Review by kresimir-coko — APPROVED** [review]

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**